### PR TITLE
Refactor build process using `next-mdx-remote`

### PR DESCRIPTION
**Pull request by katamartin**
This PR overhauls the build process used to establish routing for each blog post. Instead of using a `build-pages` script to write files in `pages/blog/`, a dynamic `pages/blog/[id].js` is defined, which uses `next-mdx-remote` to load MDX dynamically via `getStaticProps`.

A couple of changes to point out:
- Use `gray-matter` to parse metadata stored in markdown-style frontmatter
- Requires configuring React components that will be used by any post MDX in a single `components` object and threading through to `MDXRemote`
  - Dynamically import post-specific components (i.e. figures)
  - Consistently import standard components like `Box`, `Figure`, etc.

### CODE REVIEW

File: utils/mdx-utils.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const fs = require('fs')
+const path = require('path')
+const matter = require('gray-matter')
+
+// POSTS_PATH is useful when you want to get the path to a specific file
+const POSTS_PATH = path.join(process.cwd(), 'posts')
```
Comment (andersy005) [inline utils/mdx-utils.js:6]: use the parent directory of `mdx-utils.js` file regardless of where the script is invoked from?

```suggestion
const POSTS_PATH = path.join(path.dirname(__dirname), 'posts')
```

**Review by andersy005 — COMMENTED** [review]
@katamartin, i left a minor suggestion below. and thank you for putting this together. i was able to use this over in xarray.dev



do we still need these two

https://github.com/carbonplan/blog/blob/c9c9b7d86ab474cd365821cc821667ba4cac6907/package.json#L60-L61

File: utils/mdx-utils.js (hunk 1 above)
Comment (katamartin) [reply utils/mdx-utils.js:8]: And hm at least locally `path.dirname(__dirname)` returns `blog/.next/server`, which breaks routing. I'm not too familiar with best practices here, I frankly just copied over `process.cwd()` from a [Next example](https://github.com/vercel/next.js/blob/5907e9d3945d6471438cc087e652ca244e3d3c8e/examples/with-mdx-remote/utils/mdxUtils.js#L5) 😅

Comment (andersy005) [reply utils/mdx-utils.js:8]: good catch. my python bias made me believe this behaved the same way and it turns out that using `path.join(path.dirname(__dirname), "src/posts")` was actually breaking my `next` builds. I reverted back to using `process.cwd()` and everything seems to be working fine.

**Comment by freeman-lab** [comment]
confirming that this lgtm! huge work here, and really excited both for the cleanup and for what it will enable

**Pull request merged**